tests/integration/models/device.spec.coffee

```diff
@@ -1028,6 +1028,25 @@ describe 'Device Model', ->
 					m.chai.expect(_.find(result, { name: 'A' }).value).equal('a')
 					m.chai.expect(_.find(result, { name: 'B' }).value).equal('b')
 
+		describe 'balena.models.device.getSupervisorTargetState()', ->
+
+			it 'should be rejected if the device does not exist', ->
+				promise = balena.models.device.getSupervisorTargetState('asdfghjkl')
+				m.chai.expect(promise).to.be.rejectedWith('Device not found: asdfghjkl')
+
+			it 'should reflect the device\'s name', ->
+				balena.models.device.getSupervisorTargetState(@device.id).then (state) ->
+					m.chai.expect(state.local.name).to.be.a('string')
```

Comment:
In general for this and the other test case, it would be great if we can test for something more concrete than the type of a field. Since we know the name of the device, we should check if the name matches, not just that it is a string. In the other test cases as well, if some of the values are predictable, it is better to test against concrete values to avoid false positives.

Reply:
@sradevski here's the full response from the API in this case:

```
{"local":{"name":"icy-moon","config":{"RESIN_SUPERVISOR_NATIVE_LOGGER":"true","RESIN_SUPERVISOR_POLL_INTERVAL":"900000"},"apps":{"1424972":{"name":"FooBar","services":{},"volumes":{},"networks":{}}}},"dependent":{"apps":{},"devices":{}}}
```

I have added more exact checks (device name, app name, config options) however we will need to do a fair bit more leg work to set up volumes or networks and confirm that they have been returned properly.

Reply:
If it requires too much work, it might not be necessary. It was just my initial impression, but not something that is required.